### PR TITLE
Bail tests on test fixture setup error

### PR DESCRIPTION
**Pull request by jjngx**
### Proposed changes

- Test assertions are not executed if required test fixtures report errors
- Mark explicitly setup funcs as test helpers

### Checklist
Before creating a PR, run through this checklist and mark each as complete.

- [X] I have read the [CONTRIBUTING](https://github.com/nginxinc/kubernetes-ingress/blob/main/CONTRIBUTING.md) doc
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have checked that all unit tests pass after adding my changes
- [ ] I have updated necessary documentation
- [X] I have rebased my branch onto main
- [X] I will ensure my PR is targeting the main branch and pulling from my branch from my own fork

### CODE REVIEW

**Review by lucacome — APPROVED** [review]
🔥

File: internal/configs/configurator_test.go [hunk 1]
```diff
@@ -29,58 +29,51 @@ func createTestStaticConfigParams() *StaticConfigParams {
...
-		return nil, err
+		t.Fatal(err)
 	}
 
 	manager := nginx.NewFakeManager("/etc/nginx")
-
 	cnf, err := NewConfigurator(manager, createTestStaticConfigParams(), NewDefaultConfigParams(false), templateExecutor, templateExecutorV2, false, false, nil, false, nil, false), nil
 	if err != nil {
-		return nil, err
+		t.Fatal()
```
Comment (lucacome) [inline internal/configs/configurator_test.go:47]: we don't want to show the error here?

File: internal/configs/configurator_test.go [hunk 2]
```diff
@@ -29,58 +29,51 @@ func createTestStaticConfigParams() *StaticConfigParams {
...
-		return nil, err
+		t.Fatal(err)
 	}
 
 	manager := nginx.NewFakeManager("/etc/nginx")
-
 	cnf, err := NewConfigurator(manager, createTestStaticConfigParams(), NewDefaultConfigParams(false), templateExecutor, &version2.TemplateExecutor{}, false, false, nil, false, nil, false), nil
 	if err != nil {
-		return nil, err
+		t.Fatal()
```
Comment (lucacome) [inline internal/configs/configurator_test.go:68]: same as the other comment

File: internal/configs/configurator_test.go (hunk 1 above)
Comment (jjngx) [reply internal/configs/configurator_test.go:47]: good catch, fixing now

**Review by vepatel — APPROVED** [review]

**Pull request merged**